### DISCUSSION TITLE
Is there an EFI configuration file?

### DISCUSSION THREAD

**Discussion by timkgh** [zbm-dev/zfsbootmenu#578] [Questions] [answered]
Looking at a system set up with Grub, there's `/boot/efi/EFI/ubuntu/grub.cfg` next to `grubx64.efi`

Is a similar file available or necessary for ZBM EFI or is the only way to configure it by editing the `.cmdline` section of the ZBM EFI executable, using `zbm-kcl`?

If there is no config file, what is the reasoning behind it? I think it would be easier to just edit a config file than run the tool to modify the executable.

**Comment by classabbyamp** [comment]
`zbm-kcl` is the way to edit the commandline for a zbm EFI, yes

there is a configuration file that stores the commandline for zbm if you're buildling your own images, but keeping everything self-contained in the ESP is desirable feature for us (this also means it's very portable - you only have to keep track of 1 file)

**Comment by zdykstra** [comment, marked as answer]
The ZFSBootMenu core is just a Linux kernel. We've chosen to use the kernel command line (KCL) as the way to configure the behavior of ZFSBootMenu. For the pre-built EFI releases, the EFI stub loader looks to the .cmdline section in the bundle for that configuration. Otherwise if you use `rEFInd` it can pass arbitrary command lines in to the EFI bundle or to a bare kernel/initramfs.

Configuration files in the same vein as grub.cfg on an ESP are not planned.

**Comment by timkgh** [comment]
It makes sense, though it makes upgrades a little different from what people may be used to. Instead of just downloading and copying a new EFI file and keeping an existing config file, the EFI needs to first be modified (using `zbm-kcl`).
